testcases/ock_tests.sh.in

```diff
@@ -48,13 +63,13 @@ NONEED_TOKEN_INIT=0
 # login/login MUST come last if it appears in this list
 #
 OCK_TESTS="crypto/*tests"
-OCK_TESTS+=" pkcs11/attribute pkcs11/copyobjects pkcs11/destroyobjects"
-OCK_TESTS+=" pkcs11/findobjects pkcs11/generate_keypair"
-OCK_TESTS+=" pkcs11/get_interface pkcs11/getobjectsize pkcs11/sess_opstate"
-OCK_TESTS+=" misc_tests/fork misc_tests/obj_mgmt_tests" 
-OCK_TESTS+=" misc_tests/obj_mgmt_lock_tests misc_tests/reencrypt"
-OCK_TESTS+=" misc_tests/events misc_tests/cca_export_import_test"
-OCK_TESTS+=" misc_tests/dual_functions"
+OCK_TESTS="$OCK_TESTS pkcs11/attribute pkcs11/copyobjects pkcs11/destroyobjects"
```

Comment:
are these changes because `sh` does not support `+=` ?
I guess we should then change it in the Makefile.am to use `/bin/bash` instead of `sh`:

```
installcheck-local: all
	killall -HUP pkcsslotd || true
	@sbindir@/pkcsslotd
	if test ! -z ${PKCS11_TEST_USER}; then				\
		chmod 777 ${srcdir}/testcases &&			\
		cd ${srcdir}/testcases &&                               \
		su ${PKCS11_TEST_USER} -s /bin/bash -c "PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so OCK_COMBINED_EXTRACT=$(OCK_COMBINED_EXTRACT) /bin/bash ./ock_tests.sh || true"; \
	else								\
		cd ${srcdir}/testcases && 				\
		PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so OCK_COMBINED_EXTRACT=$(OCK_COMBINED_EXTRACT) /bin/bash ./ock_tests.sh || true; \
	fi
	killall -HUP pkcsslotd

```

Reply:
I considered that, but the problem is that AIX only ships with `ksh` as the default shell. `bash` must be installed through AIX Toolbox, which requires `rpm` and friends to be installed automatically.

I thought I'd best avoid this dependency ballooning and instead change the source to not be bash-specific wherever possible. Of course, I am using `bash` on my current AIX build machine, but not everyone might be ready to install a new shell just for compiling this project.

But then again, `p11sak_test.sh` and `pkcsconf_test.sh` both explicitly use `bash`-isms, so I guess we might as well embrace it?

Reply:
The makefile also uses /bin/bash at some places explicitly, so I guess it won't work without bash anyway.